### PR TITLE
Created a responsive header

### PR DESCRIPTION
**Pull request by Weeezzt**
This PR introduces a responsive `Header` component featuring:
- A desktop navigation bar with links and a login button.
- A mobile menu with a toggle functionality, utilizing the `useState` hook to control visibility.
- Conditional rendering and smooth transitions for the mobile menu for a seamless user experience.
- Integration of `react-icons` for a visually appealing and compact menu icon (`FaBars`).

**Key Features:**
1. **Desktop Navigation:** Visible on larger screens (`md:flex`), includes links for "Home," "About," "Contact," and a "Login" button.
2. **Mobile Navigation:** A collapsible menu triggered by the hamburger icon, optimized for smaller screens (`md:hidden`).
3. **Styling:** Uses Tailwind CSS for consistent and responsive design across devices.
4. **Avatar Placeholder:** A rounded avatar image (`/bild.png`) included for profile representation.

**Testing:**
- Verified functionality on various screen sizes.
- Checked hover and active states for navigation links.
- Ensured smooth transitions and animations for the mobile menu.

**Future Enhancements:**
- Replace hardcoded `href="#"` with dynamic routes.
- Add user authentication states for the "Login" button and avatar display.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Header` component for navigation, featuring links and a mobile menu.
	- Added `MenuItems` and `MenuItemsMobile` components for structured navigation.
	- Updated the application layout for a cleaner presentation.

- **Style**
	- Transitioned to Tailwind CSS for styling, replacing previous custom styles.

- **Chores**
	- Updated project dependencies to include `autoprefixer`, `postcss`, `react-icons`, and `tailwindcss`.

- **Bug Fixes**
	- Removed unused CSS and interactive elements to streamline the application.

### CODE REVIEW

**Review by MickeMarten — APPROVED** [review]
The code is very easy to read, it's well structured and the responsiveness is on point.
The dropdown animation is smooth.

A thought is to use typescript for future development.

Good job!

File: frontend/src/components/common/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  return (
+    <>
+      <header className="w-full bg-white border-b-2 shadow-sm relative z-20 h-16 px-4">
+        <div className="flex justify-between h-full ">
+          <div className="h-full flex items-center">
+            <h1 className="text-2xl font-semibold font-serif">LOGO</h1>
+          </div>
+          <div className="flex flex-row align-center h-full">
+            <div className="hidden md:flex items-center h-full mr-10">
+              <nav className="h-full w-full font-semibold">
```
Comment (MatsJoh50) [inline frontend/src/components/common/Header.tsx:20]: While this is a good start I would rather see a refactoring of the nav => ul => li for easyer editing of the menu, same for the mobile.
code ex below: 

const menuItems: { title: string, url: string }[] = [
        { title: "Home", url: "#" },
        { title: "About", url: "#" },
        { title: "Contact", url: "#" },
        { title: "Tester", url: "#"}
    ];

    const ProfileButton = () => {
        return (
            <li className="h-full flex items-center">
                <button className="p-1 w-24 bg-blue-600 text-white rounded-md shadow-md hover:bg-blue-700">
                    <p className="text-lg">Login</p>
                </button>
            </li>
        );
    };

    const RenderMenuItems = () => {
        return (
            <nav className="h-full w-full font-semibold">
                <ul className="h-full flex gap-8">
                    {menuItems.map((menu, index) => (
                        <li
                            key={index}
                            className="h-full flex items-center border-b-2 border-white hover:border-purple-800"
                        >
                            <a href={menu.url} className="text-black text-xl">
                                {menu.title}
                            </a>
                        </li>
                    ))}
                    <ProfileButton/>
                </ul>
            </nav>
        );
    };

**Review by MatsJoh50 — CHANGES_REQUESTED** [review]
Great start for a "blank" header!

I dropped a suggestion on line 20 with code example tha I would like to see fixed in similar way.

With the "Auto updated" header based on MenuItems instead of hard coded link it will be easyer to maintain in the future, both with styiling since we only need to edit the TW at one place.

please implement the same on the mobile menu.

**Review by AlbinNils98 — DISMISSED** [review]
As @MatsJoh50 mentioned i also think it's good practice to render the li items through an array of objects instead of hard coded in the tsx, it makes it easier to read and maintain. Other than that i think this PR can be merged. Great job!

File: frontend/src/components/common/Header.tsx (hunk 1 above)
Comment (AlexanderK88) [reply frontend/src/components/common/Header.tsx]: We have implemented the requested change

**Comment by AlexanderK88** [comment]
> As @MatsJoh50 mentioned i also think it's good practice to render the li items through an array of objects instead of hard coded in the tsx, it makes it easier to read and maintain. Other than that i think this PR can be merged. Great job!

We have implemented the requested change

**Review by MikaelaMathisson — APPROVED** [review]
Looks great!

**Pull request merged**